pkg/controller/secretannotator/azure/reconciler_test.go

```diff
@@ -23,20 +26,99 @@ import (
 
 	ccazure "github.com/openshift/cloud-credential-operator/pkg/azure"
 	. "github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/azure"
-	annotatorconst "github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/constants"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/azure/mock"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/constants"
 )
 
 const (
 	testNamespace = "test"
 )
 
+const (
+	TestResource                = "SomeResource"
+	TestClientID                = "SomeClientID"
+	TestClientSecret            = "SomeClientSecret"
+	TestTenantID                = "SomeTenantID"
+	TestActiveDirectoryEndpoint = "https://login.test.com/"
+)
+
+var (
+	testOAuthConfig, _ = adal.NewOAuthConfig(TestActiveDirectoryEndpoint, TestTenantID)
+	TestOAuthConfig    = *testOAuthConfig
+)
+
 func TestAzureSecretAnnotatorReconcile(t *testing.T) {
 	apis.AddToScheme(scheme.Scheme)
 	configv1.Install(scheme.Scheme)
-	existingSecret := []runtime.Object{&corev1.Secret{
+
+	tests := []struct {
+		name  string
+		wants func(*corev1.Secret)
+		roles []string
+	}{
+		{
+			name: "mint mode",
+			wants: func(s *corev1.Secret) {
+				s.Annotations[constants.AnnotationKey] = constants.MintAnnotation
+			},
+			roles: []string{"Application.ReadWrite.OwnedBy"},
+		},
+		{
+			name: "passthrough mode",
+			wants: func(s *corev1.Secret) {
+				s.Annotations[constants.AnnotationKey] = constants.PassthroughAnnotation
+			},
+			roles: []string{"Application.ReadWrite"},
+		},
+		{
+			name: "invalid credentials",
+			wants: func(s *corev1.Secret) {
+				s.Annotations[constants.AnnotationKey] = constants.InsufficientAnnotation
+			},
+			roles: nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			base := getInputSecret()
+			fakeClient := fake.NewFakeClient(base)
+			mockCtrl := gomock.NewController(t)
+			mockAdalClient := mock.NewMockAdalClient(mockCtrl)
+			setupAdalMock(mockAdalClient.EXPECT(), test.roles)
+			rcc := &ReconcileCloudCredSecret{
+				Client: fakeClient,
+				Logger: log.WithField("controller", "testController"),
+				Adal:   mockAdalClient,
+			}
+
+			// error will end-up in InsufficientAnnotation on the secret
+			rcc.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "azure-credentials",
```

Comment:
can we constify the "azure-credentials" string into something like testSecretName so we can use the var in the three places it appears in this file?